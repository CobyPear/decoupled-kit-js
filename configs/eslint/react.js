/** @type {import('eslint').ESLint.ConfigData} */
module.exports = {
	plugins: ['react'],
	extends: [
		'./index.js',
		'plugin:react/recommended',
		'plugin:react-hooks/recommended',
		'plugin:react/jsx-runtime',
	],
	settings: {
		react: {
			version: 'detect',
		},
	},
};
