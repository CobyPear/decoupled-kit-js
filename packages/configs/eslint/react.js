module.exports = {
	plugins: ['react'],
	extends: [
		'./index.js',
		'plugin:react/recommended',
		'plugin:react-hooks/recommended',
	],
	settings: {
		react: {
			version: 'detect',
		},
	},
};
