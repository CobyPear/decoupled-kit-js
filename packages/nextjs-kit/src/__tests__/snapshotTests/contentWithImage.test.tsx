import { render } from '@testing-library/react';
import ContentWithImage from '../../components/contentWithImage';
import examplePostData from '../data/examplePostData.json';
import { vi } from 'vitest';

vi.mock('../../__mocks__/next/image');

vi.mock('next/router', () => ({
	useRouter: () => ({
		locale: 'en',
	}),
}));

/**
 * @vitest-environment jsdom
 */

describe('<ContentWithImage />', () => {
	it("should render 'contentWithImage'", () => {
		const { asFragment } = render(
			<ContentWithImage
				title={examplePostData[0].title}
				content={examplePostData[0].content}
				imageProps={examplePostData[0].featuredImage}
				date={examplePostData[0].date}
			/>,
		);

		expect(asFragment()).toMatchSnapshot();
	});
});
