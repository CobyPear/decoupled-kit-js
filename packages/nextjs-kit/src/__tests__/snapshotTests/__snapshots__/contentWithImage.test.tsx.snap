// Vitest Snapshot v1

exports[`<ContentWithImage /> > should render 'contentWithImage' 1`] = `
<DocumentFragment>
  <article
    class="prose lg:prose-xl mt-10 mx-auto"
  >
    <h1>
      Example Post with Image
    </h1>
    <p
      class="text-sm text-gray-600"
    >
      Fri Jun 17 2022
    </p>
    <a
      class="font-normal"
      href="/"
    >
      Home â†’
    </a>
    <div
      class="mt-12 max-w-lg mx-auto lg:grid-cols-3 lg:max-w-screen-lg"
    >
      <div
        class="relative w-full rounded-lg shadow-lg overflow-hidden mb-10"
        style="height: 50vh;"
      >
        <span
          style="box-sizing: border-box; display: block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px;"
        >
          <img
            alt="https://dev-decoupled-wordpress-qa.pantheonsite.io/wp-content/uploads/2022/08/pizza.jpeg"
            data-nimg="fill"
            decoding="async"
            sizes="100vw"
            src="/_next/image?url=https%3A%2F%2Fdev-decoupled-wordpress-qa.pantheonsite.iohttps%3A%2F%2Fdev-decoupled-wordpress-qa.pantheonsite.io%2Fwp-content%2Fuploads%2F2022%2F08%2Fpizza.jpeg&w=3840&q=75"
            srcset="/_next/image?url=https%3A%2F%2Fdev-decoupled-wordpress-qa.pantheonsite.iohttps%3A%2F%2Fdev-decoupled-wordpress-qa.pantheonsite.io%2Fwp-content%2Fuploads%2F2022%2F08%2Fpizza.jpeg&w=640&q=75 640w, /_next/image?url=https%3A%2F%2Fdev-decoupled-wordpress-qa.pantheonsite.iohttps%3A%2F%2Fdev-decoupled-wordpress-qa.pantheonsite.io%2Fwp-content%2Fuploads%2F2022%2F08%2Fpizza.jpeg&w=750&q=75 750w, /_next/image?url=https%3A%2F%2Fdev-decoupled-wordpress-qa.pantheonsite.iohttps%3A%2F%2Fdev-decoupled-wordpress-qa.pantheonsite.io%2Fwp-content%2Fuploads%2F2022%2F08%2Fpizza.jpeg&w=828&q=75 828w, /_next/image?url=https%3A%2F%2Fdev-decoupled-wordpress-qa.pantheonsite.iohttps%3A%2F%2Fdev-decoupled-wordpress-qa.pantheonsite.io%2Fwp-content%2Fuploads%2F2022%2F08%2Fpizza.jpeg&w=1080&q=75 1080w, /_next/image?url=https%3A%2F%2Fdev-decoupled-wordpress-qa.pantheonsite.iohttps%3A%2F%2Fdev-decoupled-wordpress-qa.pantheonsite.io%2Fwp-content%2Fuploads%2F2022%2F08%2Fpizza.jpeg&w=1200&q=75 1200w, /_next/image?url=https%3A%2F%2Fdev-decoupled-wordpress-qa.pantheonsite.iohttps%3A%2F%2Fdev-decoupled-wordpress-qa.pantheonsite.io%2Fwp-content%2Fuploads%2F2022%2F08%2Fpizza.jpeg&w=1920&q=75 1920w, /_next/image?url=https%3A%2F%2Fdev-decoupled-wordpress-qa.pantheonsite.iohttps%3A%2F%2Fdev-decoupled-wordpress-qa.pantheonsite.io%2Fwp-content%2Fuploads%2F2022%2F08%2Fpizza.jpeg&w=2048&q=75 2048w, /_next/image?url=https%3A%2F%2Fdev-decoupled-wordpress-qa.pantheonsite.iohttps%3A%2F%2Fdev-decoupled-wordpress-qa.pantheonsite.io%2Fwp-content%2Fuploads%2F2022%2F08%2Fpizza.jpeg&w=3840&q=75 3840w"
            style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%; object-fit: cover;"
          />
        </span>
      </div>
    </div>
    <div
      class="break-words mt-12"
    >
      <p>
        Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua.
      </p>
      



      <blockquote
        class="wp-block-quote"
      >
        <p>
          a Quote
        </p>
        <cite>
          from QA
        </cite>
      </blockquote>
      




      <figure
        class="wp-block-pullquote"
        style="border-color:#0073a8"
      >
        <blockquote
          class="has-text-color has-dark-gray-color"
        >
          <p>
            a pull quoteeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeee
          </p>
          <cite>
            from qa
          </cite>
        </blockquote>
      </figure>
      




      <p>
        An
        <strong>
          other block
        </strong>
      </p>
      

    </div>
  </article>
</DocumentFragment>
`;
